make_group_file.py

```diff
@@ -285,13 +285,15 @@ def make_group_file(
     rows_matching = final_df[final_df.iloc[:, 0] == block_id]
 
     # Get the row index of the first matching row
-    start_idx = rows_matching.index[0]
-
-    block_df = final_df.loc[start_idx : start_idx + 2].copy()
-    block_df = block_df.dropna(axis=1)
-
-    with group_file.open('w') as gdf:
-        block_df.to_csv(gdf, index=False, header=False, sep=' ')
+    if not rows_matching.empty:
+        start_idx = rows_matching.index[0]
+        block_df = final_df.loc[start_idx : start_idx + 2].copy()
+        block_df = block_df.dropna(axis=1)
+        with group_file.open('w') as gdf:
+            block_df.to_csv(gdf, index=False, header=False, sep=' ')
+    else:
+        print(f"Block '{block_id}' not found.")
+        exit()
```

Comment:
```suggestion
```

Personally I'd dump the `exit` here. It's the last statement in the method, and the method is called in it's own job. If you do nothing, it'll just end without producing any output file.

`exit` is an alias for `quit`, which throws an error on the way out. That could cancel other jobs in the batch. Probably not given that there are no dependencies, but it would mean the batch has a mix of successful and failed jobs